practice/rain.py

```diff
@@ -2,129 +2,135 @@
 """
 # setup:
-filename = 'sunnyside.txt'
+FILENAME = 'sunnyside.txt'
 
-
-def each_line_of_file(filename):
+def open_file_and_read_lines(filename):
     """Opens sunnyside.txt and reads it line by line."""
-    with open(filename, newline='') as f:
+    with open(FILENAME, newline='') as f:
         lines = f.readlines()
+    header_removed = lines[11:-1]
 
-    return lines
-
+    return header_removed
 
-def split_lines(lines):
-    r"""Strips and splits the lines into lists.
 
-    >>> split_lines(['1 \n', '2\n', '3\n', '4\n', '5\n', '6 \n', '7 \n', '8 \n', '9 \n', '10 \n', '11 \n', 'PARTY! \n', 'TIME!'])
-    [['PARTY!']]
+def parse_relevant_lines_into_date_amount(lines):
     """
-    listed_lines = [x.strip().split('\n') for x in lines]
-    listed_lines_reduced = listed_lines[11:-1]
-    line_list = []
-    for listed_lines_split in listed_lines_reduced:
-        line_list += [y.split() for y in listed_lines_split]
+    >>> parse_relevant_lines_into_date_amount(['01-JAN-9999 0 1 2 3 4', '02-JAN-9999 5 6 7 8 9', '03-JAN-9999 - - - - -'])
+    [('01-JAN-9999', 0), ('02-JAN-9999', 5)]
+    """
+    return [parse_line_into_date_amount(line) for line in lines if should_process_input_line(line)] # L
 
-    return line_list
 
+def should_process_input_line(line): # keep this separated for now
+    """
+    >>> should_process_input_line('02-JAN-9999 5 6 7 8 9')
+    True
+    >>> should_process_input_line('03-JAN-9999 - - - - -')
+    False
+    """
+    return line.split()[1] != '-' # this one line can fo to where L is
 
-def remove_junk(line_list):
-    """Cleans up the list and sets daily rain amount to an int.
 
-    >>> remove_junk([['test', '1', '2', '3']])
-    [['test', 1]]
+def parse_line_into_date_amount(line):
+    """
+    >>> parse_line_into_date_amount('01-JAN-9999 0 1 2 3 4')
+    ('01-JAN-9999', 0)
     """
-    temp_list = []
-    for z in line_list:
-        temp_list += [z[0:2]]
-    temp_list2 = []
-    for date, rain in temp_list:
-        if rain != '-':
-            int_rain = int(rain)
-            temp_list2 += [[date, int_rain]]
-    return temp_list2
-
-
-def dict_dates(y):
+    date_amount_tuple = tuple(line.split()[0:2])
+    date_amount_tuple_int = date_amount_tuple[0], int(date_amount_tuple[1])
+    return date_amount_tuple_int
+
+def create_dict_from_list_of_dates_and_rain(y):
     """Creates a dict from the list.
 
-    >>> dict_dates([['test', 1]])
-    {'test': 1}
+    >>> create_dict_from_list_of_dates_and_rain([('12-MAY-1999', 3)])
+    {'12-MAY-1999': 3}
     """
     d = {key: value for (key, value) in y}
-    d = {str(k): int(v) for k, v in d.items()}
     return d
 
 
-def most_rain(d):
+def most_rain(p):
     """Returns the key for max of dict daily totals.
 
-    >>> most_rain({'test': 1})
-    'test'
+    >>> most_rain([('10-DEC-1988', 1), ('14-JAN-1999', 10)])
+    '14-JAN-1999'
     """
-    day_with_most_rain = max(d, key=d.get)
+    dict_of_parsed_lines = create_dict_from_list_of_dates_and_rain(p)
```

Comment:
I decided to use a dict so I could easily get the max value and return the key to be used to print  a legible output. When I tried doing this with a list, it was somewhat messy.